src/Shared/Model/Repository.cs

```diff
@@ -0,0 +1,159 @@
+        public string? Type { get; set; }
+
+        [JsonProperty(PropertyName = "updated_at", NullValueHandling = NullValueHandling.Ignore)]
+        public DateTimeOffset? UpdatedAt { get; set; }
+
+        [JsonProperty(PropertyName = "url", NullValueHandling = NullValueHandling.Ignore)]
+        public string? Uri { get; set; }
+
+        public async Task<CSTRepository?> GetGithubRepositoryMetadata(PackageURL purl)
+        {
+            CSTRepository repository = new CSTRepository();
+            Purl = purl.ToString();
+
+            if (purl.Type != "github")
+            {
+                Logger.Warn("Only github repos are handled currently");
+                return repository;
+            }
+            try
+            {
+                var github = new GitHubClient(new ProductHeaderValue("OSSGadget"));
+                var ghRepository = await github.Repository.Get(purl.Namespace, purl.Name);
+
+                if (ghRepository is null) { return null; }
+                Archived = ghRepository.Archived;
+                CreatedAt = ghRepository.CreatedAt;
+                UpdatedAt = ghRepository.UpdatedAt;
+                Description = Utilities.GetMaxClippedLength(ghRepository.Description);
+                IsFork = ghRepository.Fork;
+                Forks = ghRepository.ForksCount;
+                Homepage = Utilities.GetMaxClippedLength(ghRepository.Homepage);
+                Id = ghRepository.Id;
+                Language = Utilities.GetMaxClippedLength(ghRepository.Language);
+                Name = ghRepository.Name;
+                OpenIssuesCount = ghRepository.OpenIssuesCount;
+                Parent = ghRepository.Parent?.Url;
+                PushedAt = ghRepository.PushedAt;
+                Size = ghRepository.Size;
+                FollowersCount = ghRepository.StargazersCount;
+                Uri = Utilities.GetMaxClippedLength(ghRepository.Url);
+                StakeholdersCount = ghRepository.WatchersCount;
+
+                if (ghRepository.License is not null)
+                {
+                    Licenses ??= new List<License>();
+                    Licenses.Add(new License()
+                    {
+                        Name = ghRepository.License.Name,
+                        Url = ghRepository.License.Url,
+                        SPIX_ID = ghRepository.License.SpdxId
+                    });
+                }
+
+                Owner ??= new User()
+                {
+                    Id = ghRepository.Owner.Id,
+                    Name = ghRepository.Owner.Name,
+                    Email = ghRepository.Owner.Email,
+                    Url = ghRepository.Owner.Url,
+                    Active = !ghRepository.Owner.Suspended
+                };
+            }
+            catch (Exception ex)
+            {
+                Logger.Warn($"Exception occurred while retrieving repository data: {ex.ToString()}");
+            }
+            return repository;
```

Comment:
I don't see anywhere in this function that you modify repository. Wont it always be empty?

Reply:
correct, should have returned 'this' instead